auth/service.cc

```diff
@@ -90,8 +98,22 @@ class auth_migration_listener final : public ::service::migration_listener {
     }
 
     void on_drop_user_type(const sstring& ks_name, const sstring& type_name) override {}
-    void on_drop_function(const sstring& ks_name, const sstring& function_name) override {}
-    void on_drop_aggregate(const sstring& ks_name, const sstring& aggregate_name) override {}
+    void on_drop_function(const sstring& ks_name, const sstring& function_name) override {
+        (void)_authorizer.revoke_all(
+            auth::make_functions_resource(ks_name, function_name)).handle_exception_type([](const unsupported_authorization_operation&) {
+            // Nothing.
```

Comment:
Just curious, why do we need to add this handle_exception_type(unsupported_authorization_operation) all over the code? When can this operation be "unsupported"? And if it's always fine that it's unsupported, why does this function need to throw an exception that all its callers need to ignore? (or are their certain callers which don't ignore it?)

Reply:
From what I can see, this is the error that is thrown when we try to grant/revoke/list permissions using the AllowAllAuthorizer, and we actually do not ignore it when executing grant/list/revoke statements - in these cases we prepare an `invalid_request_exception` instead. In the case of `grant_permissions_to_creator`, we probably ignore it because it's an internal function - otherwise we would reject any CREATE statements of the user.
That said, I'm not sure if it's the best way of implementing the AllowAllAuthorizer, but even if it isn't, it's a problem for another patch